examples/Numerics/DGmethods/nonnegative.jl

```diff
@@ -0,0 +1,258 @@
+# This example uses the TMAR Filter from
+#
+#    @article{doi:10.1175/MWR-D-16-0220.1,
+#      author = {Light, Devin and Durran, Dale},
+#      title = {Preserving Nonnegativity in Discontinuous Galerkin
+#               Approximations to Scalar Transport via Truncation and Mass
+#               Aware Rescaling (TMAR)},
+#      journal = {Monthly Weather Review},
+#      volume = {144},
+#      number = {12},
+#      pages = {4771-4786},
+#      year = {2016},
+#      doi = {10.1175/MWR-D-16-0220.1},
+#    }
+#
+# to reproduce the example in section 4b.  It is a shear swirling
+# flow deformation of a transported quantity from LeVeque 1996.  The exact
+# solution at the final time is the same as the initial condition.
+
+using MPI
+using Test
+using CLIMA
+using Logging
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.Mesh.Filters
+using CLIMA.DGmethods
+using CLIMA.DGmethods.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.ODESolvers
+using LinearAlgebra
+using Printf
+using Dates
+using CLIMA.GenericCallbacks: EveryXWallTimeSeconds, EveryXSimulationSteps
+using CLIMA.VTK: writevtk, writepvtu
+
+include(joinpath(
+    @__DIR__,
+    "..",
+    "..",
+    "..",
+    "test",
+    "Numerics",
+    "DGmethods",
+    "advection_diffusion",
+    "advection_diffusion_model.jl",
+))
```

Comment:
Unfortunately this broke the docs, as the relative path is no longer correct for the notebook generated by Literate.jl.

Reply:
Hm. Also, does it seem awkward for a tutorial to use code in `test/`?

Reply:
Yes, we should probably have a better way of doing that.

Reply:
We could make the advection diffusion a first class physics model(?)

Reply:
Is there a way in Julia to get base path of the project / git repo?

Reply:
Following `using CLIMA`: `const clima_dir = dirname(dirname(pathof(CLIMA)))`

Reply:
I really like the idea of making advection-diffusion a first class physics model, though. I think abstracting some of the models will help make them more easily extendable.

Reply:
> We could make the advection diffusion a first class physics model(?)

Alternatively, advection-diffusion could be turned in a tutorial on how to create a physics model.